rasa/nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -218,84 +220,85 @@ def required_packages(cls) -> List[Text]:
-                list(label_id_dict.keys()), self.label_split_symbol
-            )
-
-            encoded_all_label_ids = np.zeros(
-                (len(label_id_dict), len(label_id_token_dict))
+    def _create_encoded_label_ids(
+        self,
+        training_data: "TrainingData",
+        label_id_dict: Dict[Text, int],
+        attribute: Text,
+        attribute_feature_name: Text,
+    ) -> np.ndarray:
+        """Create matrix with label_ids encoded in rows as bag of words."""
+
+        encoded_all_labels = []
+        single_encoding_shape = None
+
+        for label_name, idx in label_id_dict.items():
+            label_example = self._find_example_for_label(
+                label_name, training_data.intent_examples, attribute
             )
-            for key, idx in label_id_dict.items():
-                for t in key.split(self.label_split_symbol):
-                    encoded_all_label_ids[idx, label_id_token_dict[t]] = 1
-
-            return encoded_all_label_ids
-        else:
-            return np.eye(len(label_id_dict))
+            if label_example:
+                encoded_all_labels.insert(
+                    idx, label_example.get(attribute_feature_name)
+                )
+                single_encoding_shape = label_example.get(attribute_feature_name).shape
+            else:
+                encoded_all_labels.insert(idx, np.zeros_like(single_encoding_shape))
 
-    # noinspection PyPep8Naming
-    def _create_all_Y(self, size: int) -> np.ndarray:
-        """Stack encoded_all_label_ids on top of each other
-
-        to create candidates for training examples and
-        to calculate training accuracy
-        """
-
-        return np.stack([self._encoded_all_label_ids] * size)
+        return np.array(encoded_all_labels)
 
     # noinspection PyPep8Naming
     def _create_session_data(
-        self, training_data: "TrainingData", label_id_dict: Dict[Text, int]
+        self,
+        training_data: "TrainingData",
+        label_id_dict: Dict[Text, int],
+        attribute: Text,
     ) -> "train_utils.SessionData":
         """Prepare data for training"""
 
-        X = np.stack([e.get("text_features") for e in training_data.intent_examples])
+        X = []
+        label_ids = []
+        Y = []
+        for e in training_data.intent_examples:
+            if e.get(attribute):
+                X.append(e.get(MESSAGE_VECTOR_FEATURE_NAMES[MESSAGE_TEXT_ATTRIBUTE]))
+                label_ids.append(label_id_dict[e.get(attribute)])
 
-        label_ids = np.array(
-            [label_id_dict[e.get("intent")] for e in training_data.intent_examples]
-        )
+        X = np.array(X)
+        label_ids = np.array(label_ids)
 
-        Y = np.stack(
-            [self._encoded_all_label_ids[label_id_idx] for label_id_idx in label_ids]
-        )
+        for label_id_idx in label_ids:
+            Y.append(self._encoded_all_label_ids[label_id_idx])
+
+        Y = np.array(Y)
```

Comment:
Some of the tests were failing, but it was weird because your tests passed. I'll check again.

Reply:
It would fail when training data containing no response examples are sent to ResponseSelector. This would only happen in pytests because there are a couple of tests where training of all components is tested. I'll probably add some response examples for this test.